### PR TITLE
[ticket/15954] Add safeguards to include() calls

### PR DESCRIPTION
**Pull request by kasimi**
All functions and classes that are being checked for are actually the ones being used, except for the check in ucp_groups.php: https://github.com/kasimi/phpbb/commit/75007697ae55632200eb7f2c6579026e23fe3d7d#diff-ec446c980dec86d6991a822a459adfe6R404 I couldn't find out which of the functions in `includes/functions_display.php` is needed so I picked a random one for the `function_exists()` check.

PHPBB3-15954

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

https://tracker.phpbb.com/browse/PHPBB3-15954

### CODE REVIEW

File: phpBB/includes/mcp/mcp_main.php [hunk 1]
```diff
@@ -174,7 +174,10 @@ function main($id, $mode)
 		switch ($mode)
 		{
 			case 'front':
-				include($phpbb_root_path . 'includes/mcp/mcp_front.' . $phpEx);
+				if (!function_exists('mcp_front_view'))
+				{
+					mcp_front_view($id, $mode, $action);
```
Comment (battye) [inline phpBB/includes/mcp/mcp_main.php]: Did you mean for this line to be `include($phpbb_root_path . 'includes/mcp/mcp_front.' . $phpEx);` ❓

Comment (kasimi) [reply phpBB/includes/mcp/mcp_main.php]: Indeed, thank you!

**Comment by rubencm** [comment]
Why don't use include_once or even better, require_once?

**Comment by kasimi** [comment]
Sounds good to me. Any other opinions before I go ahead and change to `require_once()`

**Comment by rubencm** [comment]
Get rid of $phpEx

**Comment by DavidIQ** [comment]
That might require some commenting from @marc1706 and/or @paul999 as there is a reason that exists.

**Comment by paul999** [comment]
The phpEx comes from back in phpBB2 when it was really common to use different extensions for different php versions. In the current times it is not that usefull anymore.

However, in my personal opinion, especially in the old code base like this part of phpBB, it is really common to use both include with a function check and $phpEx, and as such I think that should be the way to go.

**Comment by 3D-I** [comment]
I agree with @paul999 FWIW

**Pull request merged**